src/backend/allocator/dmabuf.rs

```diff
@@ -251,6 +252,112 @@ impl Dmabuf {
         let blocker = DmabufBlocker(source.signal.clone());
         Ok((blocker, source))
     }
+
+    /// Map this dmabuf readable
+    pub fn map_readable(&self) -> Result<DmabufMapping, DmabufMappingFailed> {
+        self.map(rustix::mm::ProtFlags::READ)
+    }
+
+    /// Map this dmabuf read/writeable
+    pub fn map_writable(&self) -> Result<DmabufMapping, DmabufMappingFailed> {
+        self.map(rustix::mm::ProtFlags::READ | rustix::mm::ProtFlags::WRITE)
+    }
+
+    /// Sync access to this dmabuf
+    pub fn sync(&self, flags: DmaBufSyncFlags) -> std::io::Result<()> {
+        for plane in self.0.planes.iter() {
+            unsafe { rustix::ioctl::ioctl(&plane.fd, Setter::<DmaBufSync, _>::new(dma_buf_sync { flags })) }?;
+        }
+        Ok(())
+    }
+
+    fn map(&self, prot: rustix::mm::ProtFlags) -> Result<DmabufMapping, DmabufMappingFailed> {
```

Comment:
Perhaps name this `map_linear_plane` or something less generic?

Reply:
Maybe it would be a good idea to let the caller decide and provide some linear helper.

Something like:

```
pub fn map_plane(idx: usize, mode: Mode) -> Result<DmabufMapping, DmabufMappingFailed> { ... }

pub fn map_linear(mode: Mode) -> Result<DmabufMapping, DmabufMappingFailed> {
    ...checks for single plane linear
    self.map(0, mode)
}
```

sync analog to it

@ids1024 @Drakulix opinions?

Reply:
Yeah I like that idea. In case you have a tiled DumbBuffer (for whatever reason) and want to copy to another device, that can read it (or a vulkan buffer, whatever honestly), it might be useful to have that option and not be forced to only map linear buffers.

Reply:
I added more generic `map_plane` and `sync_plane` functions. The pixman renderer still only supports single plane buffers